### ISSUE TITLE
Redirect loop in creating a user session, right after a SAML login

### ISSUE THREAD

**Issue by sandorspruit** [pitbulk/limesurvey-saml#5]
I am experimenting with LimeSurvey SAML authentication, based on the plugin. Everything seems to be working just fine, users gets created in the database etc. but Yii loses the identity of the user right after the login. In particular: Yii::app()->user->getId() returns nothing after the final redirect to AdminController.

This causes LimeSurvey to redirect back and forth, indefinitely, between the AdminController (application/controllers/AdminController) and the authentication action (application/controllers/admin/authentication), until my browser gives up.

**Comment by Shnoulle** [comment]
Remind to put your LimeSurvey version :) 3.0 maybe ?

**Comment by sandorspruit** [comment]
Thanks for the heads-up :) I am on Lime 2.72.5+171121

**Comment by Shnoulle** [comment]
Did you try with deactivating CRSF ? https://github.com/pitbulk/limesurvey-saml/issues/4#issuecomment-277636362

**Comment by sandorspruit** [comment]
Yes, that was one of the first things we tried. I noticed some small changes in the authentication code, after a recent comfort update, so I'll re-investigate soon to see where that leaves me.

**Comment by Shnoulle** [comment]
This is strange, because webserver plugin are not updated since a lot of time :/

Else : maybe move this part to init : https://github.com/pitbulk/limesurvey-saml/blob/462eb5bf61d8fdeb2230ec97fad67a0dfce70791/AuthSAML/AuthSAML.php#L98

**Comment by sandorspruit** [comment]
I have already tried that. Now looking into the issue noted here:
https://simplesamlphp.org/docs/stable/simplesamlphp-sp#section_6 (last couple of lines in section 6)

**Comment by Frankniesten** [comment]
Maybe your session is lost. Try a different session storage method in simplesamlphp.

**Comment by maettle** [comment]
I've got the same problem. After klicking the SAML button I get redirected to ADFS, and after the successful login it sends me into a never ending loop (and i don't know where), FF says Redirect Error.
@sandorspruit did you solve this problem?

**Comment by sandorspruit** [comment]
Nope. At some point it was working 95% but the session issue kept coming up. Tried changing the simplesaml session storage method but it just wouldn't go away.

**Comment by foobarable** [comment]
Any hints here? Also stuck with this issue. My gut feeling is also that SSP kills the session from limesurvey and maybe the plugin needs to call $session->cleanup() (from SSP) at some point. But I still have no clue yet at what point this has to be done

**Comment by foobarable** [comment]
Yes, looks good. 

I added:             
```
$session = SimpleSAML_Session::getSessionFromRequest();
$session->cleanup();
```
to the newUserSession function right after getting all the attributes from SSP and it seems to work.
